src/Agent.Worker/Build/FileContainerServer.cs

```diff
@@ -101,8 +96,19 @@ public class FileContainerServer
 
                 try
                 {
+                    var uploadToBlob = String.Equals(context.GetVariableValueOrDefault(WellKnownDistributedTaskVariables.UploadBuildArtifactsToBlob), "true", StringComparison.InvariantCultureIgnoreCase)
+                         && !AgentKnobs.DisableBuildArtifactsToBlob.GetValue(context).AsBoolean();
+
                     // try upload all files for the first time.
-                    UploadResult uploadResult = await ParallelUploadAsync(context, files, maxConcurrentUploads, _uploadCancellationTokenSource.Token);
+                    UploadResult uploadResult;
+                    if (uploadToBlob)
+                    {
+                        uploadResult = await BlobUploadAsync(context, files, maxConcurrentUploads, _uploadCancellationTokenSource.Token);
+                    }
+                    else
+                    {
+                        uploadResult = await ParallelUploadAsync(context, files, maxConcurrentUploads, _uploadCancellationTokenSource.Token);
```

Comment:
Blob-related code was moved out of `ParallelUploadAsync`. That method now is the same as it used to be pre-blob. All blob-related code now lives in `BlobUploadAsync`